### ISSUE TITLE
npm search is broken

### ISSUE THREAD

**Issue by okket** [npm/npm#14255]
#### I'm opening this issue because:

  - [x] npm is crashing.

#### What's going wrong?

npm search

#### How can the CLI team reproduce the problem?

npm search ember

### supporting information:

Works for LTS/Argon (node 4.6.0 / npm 2.15.9) on OS X and Linux.

Crashes with similar GC errors for node 6.7.0 / npm 3.10.3 on OS X and Linux.

OS X 10.12.0:
```
<--- Last few GCs --->

   11958 ms: Mark-sweep 1010.2 (1329.2) -> 982.0 (1300.1) MB, 761.6 / 0.0 ms [allocation failure] [GC in old space requested].
   12741 ms: Mark-sweep 982.0 (1300.1) -> 981.9 (1304.1) MB, 783.5 / 0.0 ms [allocation failure] [GC in old space requested].
   13555 ms: Mark-sweep 981.9 (1304.1) -> 981.3 (1265.1) MB, 813.4 / 0.0 ms [last resort gc].
   14302 ms: Mark-sweep 981.3 (1265.1) -> 981.2 (1257.1) MB, 746.6 / 0.0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x2d77e46cfb51 <JS Object>
    1: fromString(aka fromString) [buffer.js:195] [pc=0x3915bca75030] (this=0x2d77e4604381 <undefined>,string=0x3342dc54481 <Very long string[209474054]>,encoding=0x2d77e46dd329 <String[4]: utf8>)
    2: from [buffer.js:95] [pc=0x3915bca74c62] (this=0x2c571969fa89 <JS Function Buffer (SharedFunctionInfo 0x5f5b97f63b9)>,value=0x3342dc54481 <Very long string[209474054]>,encodingOrOffset=0x2d77e46...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 4: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 5: v8::internal::String::SlowFlatten(v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 6: v8::String::WriteUtf8(char*, int, int*, int) const [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 7: node::StringBytes::Write(v8::Isolate*, char*, unsigned long, v8::Local<v8::Value>, node::encoding, int*) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 8: node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
 9: node::Buffer::CreateFromString(v8::FunctionCallbackInfo<v8::Value> const&) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
10: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
11: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
12: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*) [/Users/okketimm/.nvm/versions/node/v6.7.0/bin/node]
13: 0x3915bc4092a7
Abort trap: 6
``` 

Linux Fedora 24:
```
<--- Last few GCs --->

   15990 ms: Mark-sweep 1009.8 (1306.2) -> 982.0 (1288.1) MB, 960.7 / 0.0 ms [allocation failure] [GC in old space requested].
   17057 ms: Mark-sweep 982.0 (1288.1) -> 981.9 (1291.1) MB, 1066.2 / 0.0 ms [allocation failure] [GC in old space requested].
   18150 ms: Mark-sweep 981.9 (1291.1) -> 981.3 (1253.1) MB, 1093.2 / 0.0 ms [last resort gc].
   19240 ms: Mark-sweep 981.3 (1253.1) -> 981.2 (1246.1) MB, 1089.8 / 0.0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3e53c56cfb51 <JS Object>
    1: fromString(aka fromString) [buffer.js:195] [pc=0x1a63b0a769b0] (this=0x3e53c5604381 <undefined>,string=0x1dca4c8a4611 <Very long string[209474054]>,encoding=0x3e53c56dd329 <String[4]: utf8>)
    2: from [buffer.js:95] [pc=0x1a63b0a765e2] (this=0xb4fd8ba8391 <JS Function Buffer (SharedFunctionInfo 0x9f6b91f7661)>,value=0x1dca4c8a4611 <Very long string[209474054]>,encodingOrOffset=0x3e53c5...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [npm]
 2: 0x109838c [npm]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [npm]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [npm]
 5: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [npm]
 6: v8::internal::String::SlowFlatten(v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [npm]
 7: v8::internal::String::Flatten(v8::internal::Handle<v8::internal::String>, v8::internal::PretenureFlag) [npm]
 8: v8::String::WriteUtf8(char*, int, int*, int) const [npm]
 9: node::StringBytes::Write(v8::Isolate*, char*, unsigned long, v8::Local<v8::Value>, node::encoding, int*) [npm]
10: node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) [npm]
11: node::Buffer::CreateFromString(v8::FunctionCallbackInfo<v8::Value> const&) [npm]
12: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [npm]
13: 0x9e6a22 [npm]
14: 0x9e709e [npm]
15: 0x1a63b04092a7
```

**Comment by KenanY** [comment]
Duplicate of https://github.com/npm/npm/issues/12619. Should be fixed once https://github.com/npm/npm/pull/13746 lands.

**Comment by okket** [comment]
Thanks. Sorry for not searching deep/long enough, didn't saw the duplicate.

**Comment by KenanY** [comment]
[Fixed in `npm@4`](https://github.com/npm/npm/blob/4639b0eed8db92c8872011d4affd10c498b3b9db/CHANGELOG.md#search-rewrite-breaking)
